tiledb/sm/misc/types.h

```diff
@@ -129,6 +131,14 @@ class Range {
     return &range_[0];
   }
 
+  /** Copies 'start' into this range's start bytes for fixed-size ranges. */
+  void set_start(const void* const start) {
+    if (range_start_size_ != 0)
+      LOG_FATAL("Unexpected var-sized range.");
```

Comment:
Maybe add the something like `Cannot set start range` in the error message so that we know where this is thrown?